source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm

```diff
@@ -203,28 +204,32 @@ + (UIView *)render:(UIView<ACRIContentHoldingView> *)view
 
         [acoElem setElem:elem];
 
-        curStretchableElem = [renderer render:view rootView:rootView inputs:inputs baseCardElement:acoElem hostConfig:config];
-
-        if(elem->GetHeight() == HeightType::Stretch){
-            if(prevStretchableElem){
-                NSLayoutConstraint *heightConstraint = [NSLayoutConstraint constraintWithItem:curStretchableElem
-                                             attribute:NSLayoutAttributeHeight
-                                             relatedBy:NSLayoutRelationEqual
-                                                toItem:prevStretchableElem
-                                             attribute:NSLayoutAttributeHeight
-                                            multiplier:1
-                                              constant:0];
-                heightConstraint.priority = UILayoutPriorityDefaultLow;
-                heightConstraint.active = YES;
-            }
-
-            if([view isKindOfClass:[ACRColumnView class]]){
-                ACRColumnView *columnView = (ACRColumnView*)view;
-                columnView.hasStretchableView = YES;
+        @try {
+            curStretchableElem = [renderer render:view rootView:rootView inputs:inputs baseCardElement:acoElem hostConfig:config];
+            if(elem->GetHeight() == HeightType::Stretch){
+                if(prevStretchableElem){
+                    NSLayoutConstraint *heightConstraint = [NSLayoutConstraint constraintWithItem:curStretchableElem
+                                                                                        attribute:NSLayoutAttributeHeight
+                                                                                        relatedBy:NSLayoutRelationEqual
+                                                                                           toItem:prevStretchableElem
+                                                                                        attribute:NSLayoutAttributeHeight
+                                                                                       multiplier:1
+                                                                                         constant:0];
+                    heightConstraint.priority = UILayoutPriorityDefaultLow;
+                    heightConstraint.active = YES;
+                }
+                
+                if([view isKindOfClass:[ACRColumnView class]]){
+                    ACRColumnView *columnView = (ACRColumnView*)view;
+                    columnView.hasStretchableView = YES;
+                }
+                
+                prevStretchableElem = curStretchableElem;
             }
-
-            prevStretchableElem = curStretchableElem;
-        }
+        } @catch (NSException *e){
+            handleException(e, view, rootView, inputs, elem, config);
+            NSLog(@"Unsupported card element type:%d\n", (int) elem->GetElementType());
```

Comment:
>d\n" [](start = 51, length = 4)

Question: Who sees this error? The host app? are they able to distinguish what the numeric type is for ElementType? Would it be more "debug-friendly" to "stringify" it in the message?

Reply:
this log isn't needed since handleException() will handle the error. 

---
In reply to: [275092712](https://github.com/Microsoft/AdaptiveCards/pull/2675#discussion_r275092712) [](ancestors = 275092712)